### PR TITLE
Fix jest tests

### PR DESCRIPTION
**Pull request by sumukhswamy**
### Description
added jest tests to increase code coverage and fixed flakey tests in cypress
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Swiddis — APPROVED** [review]

File: .cypress/integration/ui.spec.js [hunk 1]
```diff
@@ -29,15 +29,15 @@ describe('Dump test data', () => {
...
+    testDataSet.forEach(({ url, index }) => dumpDataSet(url, index));
   });
 });
 
 describe('Test PPL UI', () => {
   beforeEach(() => {
     cy.visit('app/opensearch-query-workbench');
     cy.wait(delay);
-    cy.get('.euiToggle__input[title=PPL]').click({ force: true });
+    cy.get('[data-test-subj="PPL"]').click({ force: true });
```
Comment (mengweieric) [inline .cypress/integration/ui.spec.js:40]: Could we use a longer name maybe something like queryWorkbench__languageSwitcher-ppl, it helps make it more meaningful, similar for SQL appeared later in this PR

File: public/components/QueryResults/QueryResults.tsx [hunk 2]
```diff
@@ -369,7 +370,7 @@ class QueryResults extends React.Component<QueryResultsProps, QueryResultsState>
                     id="tabsContainer"
                   >
                     <EuiFlexItem style={{ marginTop: '8px' }} grow={false}>
-                      <EuiTabs style={{ marginLeft: 7, marginBottom: 6 }}>{tabsButtons}</EuiTabs>
+                      <EuiTabs data-test-subj = "result_tab" style={{ marginLeft: 7, marginBottom: 6 }}>{tabsButtons}</EuiTabs>
```
Comment (mengweieric) [inline public/components/QueryResults/QueryResults.tsx:373]: Remove spaces before and after '='?

File: test/mocks/mockData.ts [hunk 3]
```diff
@@ -2181,3 +2196,206 @@ export const mockOpenSearchTreeQuery = {
...
   },
 };
+
```
Comment (mengweieric) [inline test/mocks/mockData.ts:2199]: This file grows so large, we can do proper breakdown later but wondering if we can move these testing data to another file or even a json file. Keep adding these large piece of data will make it harder and harder to maintain

File: .cypress/integration/ui.spec.js (hunk 1 above)
Comment (sumukhswamy) [reply .cypress/integration/ui.spec.js:40]: sure

File: public/components/QueryResults/QueryResults.tsx (hunk 2 above)
Comment (sumukhswamy) [reply public/components/QueryResults/QueryResults.tsx:373]: sure

**Pull request merged**